### ISSUE TITLE
Duplicate files with unclear differences in repository

### ISSUE THREAD

**Issue by stweil** [kitodo/kitodo-production#6117]
The repository contains a lot of files which seem to exist more than once in different directories for no obvious reason.

A list of duplicate filenames can be produced with `find * -type f|perl -pi -e 's,.*/,,'|sort|uniq -c|sort -n|grep -vw 1`. Many of the duplicates are okay and expected, but some are not.

Examples of unexpected duplicates: all *.xsl, *.sh, but also kitodo_config.properties.

Duplicates make the maintainance more difficult, can lead to bugs if there are unwanted differences (for example between test and production code) and cost unnecessary storage. Ideally all duplicates should be removed, if that's not possible replaced by symbolic links.

**Comment by henning-gerhardt** [comment]
It should keep in mind that
- the tests are executed in separated Maven modules and each Maven module should need its own files
- Symbolic links did not work on every (file) system

**Comment by stweil** [comment]
That's right, symbolic won't work on NTFS. Is Windows support required? I don't know any relevant other  OS where symbolic links would be a problem.

**Comment by stweil** [comment]
If a test uses its own file(s) instead of the productive one(s), it might test the wrong things and show results which are not valid for the production code.

**Comment by henning-gerhardt** [comment]
> That's right, symbolic won't work on NTFS. Is Windows support required? I don't know any relevant other OS where symbolic links would be a problem.

So far as I know there are Windows developers.

Git is for content handling and not for file handling and symbolic links are special files. If you search about git and symbolic link you will find some problematic issues while working with symbolic links inside a git repository - especially you work with many different branches on different stages. In many cases then the symbolic link will be resolved by git and content get replaced. My reading is to not use symbolic links inside a git repository.


> If a test uses its own file(s) instead of the productive one(s), it might test the wrong things and show results which are not valid for the production code.

There will be every time a adjusted file in a productive environment which is never covered by tests so you can only test a certain range inside test execution and never the whole possible range.

**Comment by stweil** [comment]
> My reading is to not use symbolic links inside a git repository.

This is so for absolute symbolic links. When a symbolic link in a Git repository points to a file within the same repository )(like in our use case), it's generally not problematic and very common.

**Comment by henning-gerhardt** [comment]
> > My reading is to not use symbolic links inside a git repository.
> 
> This is so for absolute symbolic links. When a symbolic link in a Git repository points to a file within the same repository)(like in our use case), it's generally not problematic and very common, unless the repository must also work under Windows.

Okay, we are starting cycling (I say it is a bad idea to use symbolic links you say it is a good one) and we do not get same opinion at this point.

**Comment by stweil** [comment]
Maybe you noticed that I wrote "ideally all duplicates should be removed" and that I don't think that symlinks are a solution if "the repository must also work under Windows". So we disagree about something which is not relevant for this issue. Nevertheless I could not leave your statements without a comment because I think that they are misleading for less experienced users of Git.
